pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -1268,12 +1272,30 @@ spec:
 			"Normal Started ",
 			"Normal Running Not all Steps",
 		},
-		wantPod: expectedPod("test-taskrun-with-output-config-pod", "", "test-taskrun-with-output-config", "foo", config.DefaultServiceAccountValue, false, nil, []stepForExpectedPod{{
-			name:       "mycontainer",
-			image:      "myimage",
-			stdoutPath: "stdout.txt",
-			cmd:        "/mycmd",
-		}}),
+		wantPod: addVolumeMounts(expectedPod("test-taskrun-with-output-config-pod", "", "test-taskrun-with-output-config", "foo", config.DefaultServiceAccountValue, false,
+			[]corev1.Volume{
+				{
+					Name: spire.WorkloadAPI,
+					VolumeSource: corev1.VolumeSource{
+						CSI: &corev1.CSIVolumeSource{
+							Driver:   "csi.spiffe.io",
+							ReadOnly: &readonly,
+						},
+					},
+				}}, []stepForExpectedPod{{
+				name:       "mycontainer",
+				image:      "myimage",
+				stdoutPath: "stdout.txt",
+				cmd:        "/mycmd",
+			}}, true),
+			[]corev1.VolumeMount{
+				{
+					Name:      spire.WorkloadAPI,
+					MountPath: spire.VolumeMountPath,
+					ReadOnly:  true,
```

Comment:
why one use readonly(which is set to true), and the other uses `true`?

Reply:
CSIVolumeSource (https://pkg.go.dev/k8s.io/api/core/v1#CSIVolumeSource)'s ReadOnly type is *bool, while VolumeMount it is bool.
Ofcourse, we could use readonly in VolumeMount too.